github-git/3-fork-repo.md

```diff
@@ -0,0 +1,100 @@
+---
+jupytext:
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.16.4
+kernelspec:
+  display_name: Python 3 (ipykernel)
+  language: python
+  name: python3
+---
+
+(fork-repository)=
+# How to Fork a GitHub Repository 
```

Comment:
This is a different workflow, and so might not be easy to teach along side the lesson you have now, but if all edits happen in the browser, navigating to the file to be edited *first* and clicking the edit pen button will take the user to an alternative `Fork this repository` page (only if you haven't yet forked).

Reply:
OH gosh - so true. I discovered this when creating the screen captures! 

What is so hard is that there are actually multiple potential paths here! I thought about an overview page (i have a [draft online here](https://www.pyopensci.org/lessons/github-git/ways-to-contribute.html#start-here-if-you-haven-t-yet-forked-the-repository) that also shows them the various paths in case they happen to do what you just mentioned! 

But we stick with "first principles," so to speak, in this lesson, which would be the traditional issue, fork, work, commit, pr route. 

I want to show them one way clearly. But then maybe have a resource page that shows alternative paths? 

what do you think about that?

Reply:
Probably best to stick to one workflow. And what you have is the more general, probably seen as more "proper" as it is how you would make a fix locally.

Best to keep it simple. I'll just note that every PyOS sprint I have gone to, I've encouraged attendees to go straight for the edit button.

Reply:
Yup. ok cool. yes i think we can in the future show them those workflows as well (i've done that too in sprints). 
i've started this page below:

https://www.pyopensci.org/lessons/github-git/ways-to-contribute.html

to capture the various ways you can create a pr! That way we have that documented for sprints too (we can work on it together via a future review as well). i plan to open more pr's here with a fresh version of each lesson so we can review it as if it's new after i get this grant deliverable in today (its done!).